lib/jekyll/configuration.rb

```diff
@@ -259,10 +264,11 @@ def backwards_compatibilize
     def fix_common_issues
       config = clone
 
-      if config.key?('paginate') && (!config['paginate'].is_a?(Integer) || config['paginate'] < 1)
-        Jekyll.logger.warn "Config Warning:", "The `paginate` key must be a" \
-          " positive integer or nil. It's currently set to '#{config['paginate'].inspect}'."
-        config['paginate'] = nil
+      if config.key?("paginate") && (!config["paginate"].is_a?(Integer) ||
+          config["paginate"] < 1)
```

Comment:
For split line ifs, go as deep as the first word after if and then two deeper by whole tabs and then add an extra line after that so that it's cleaner to read.

```ruby
if config.key?("paginate") && (!config["paginate"].is_a?(Integer) ||
       config["paginate"] < 1)

  # Code Here.
end
```

It's much more readable.